### PR TITLE
iptables: enable TCP matching with "-m tcp"

### PR DESCRIPTION
**Pull request by kevinGC**
A couple other things that changed:

- There's a proper extension registration system for matchers. Anyone
  adding another matcher can use tcp_matcher.go or udp_matcher.go as
  templates.
- All logging and use of syserr.Error in the netfilter package happens at the
  highest possible level (public functions). Lower-level functions just
  return normal, descriptive golang errors.

### CODE REVIEW

File: pkg/abi/linux/netfilter.go [hunk 1]
```diff
@@ -348,6 +348,58 @@ func goString(cstring []byte) string {
...
+
+	// SourcePortEnd specifies the inclusive end of the range of source ports
+	// to which the matcher applies.
+	SourcePortEnd uint16
+
+	// DestinationPortStart specifies the start of the destination port
+	// range to which the matcher applies.
+	DestinationPortStart uint16
+
+	// DestinationPortEnd specifies the start of the destination port
```
Comment (nybidari) [inline pkg/abi/linux/netfilter.go]: The comment should be "specifies the end of the destination port"

File: pkg/sentry/socket/netfilter/extensions.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The gVisor Authors.
```
Comment (nybidari) [inline pkg/sentry/socket/netfilter/extensions.go]: Nit: 2020

Comment (kevinGC) [reply pkg/sentry/socket/netfilter/extensions.go]: Done.

File: pkg/abi/linux/netfilter.go (hunk 1 above)
Comment (kevinGC) [reply pkg/abi/linux/netfilter.go]: Done.

**Pull request merged**